### DISCUSSION TITLE
😭 please. help me! why don't do the postgrest function call

### DISCUSSION THREAD

**Discussion by OJKSmITh** [PostgREST/postgrest#3189] [Q&A] [answered]
Hello, I'm a newbie currently using postgrest.

I'm trying to debug an error that's occurring, but I can't seem to figure out why it's not working. I would greatly appreciate any help.

I've created a simple function which works perfectly in the database itself.

```
CREATE OR REPLACE FUNCTION public.dashboard_find_orders()
 RETURNS TABLE(id integer, step character varying, order_number character varying, order_item_type character varying, instruction_type character varying, work_order_id integer, work_created_at timestamp without time zone, instruction_number character varying, work_step character varying)
 LANGUAGE plpgsql
AS $function$
declare
    begin
        RETURN QUERY 
        SELECT 
            o.id, o.step, o.order_number, CAST(o.order_item_type AS character varying), 
            cast(wi.instruction_type AS character varying), wi.order_id, 
            wi.created_at, wi.instruction_number, 
            wi.step
        FROM orders o 
        LEFT JOIN work_instruction wi ON o.id = wi.order_id 
        WHERE o.company_id = 75
        ORDER BY o.id DESC;
    END
$function$
;
```

```
   SELECT * FROM public.dashboard_find_orders(); 
(The table is being returned)
```
However, when I make a request to {{server}}/rpc/dashboard_find_orders in postgrest, it only returns [].

I want to receive the table, but I'm not sure what the problem is.

Could you tell me what might be the issue? (I have already granted function execution permissions to the user, and I have definitely reloaded the server)

I would be very grateful for any help. Thank you.

**Comment by OJKSmITh** [comment, marked as answer]
다른 친절한 분께서 도움을 주셨습니다. 

문제는 바로 규약이였습니다. 테이블마다 자체 규약이 걸려있어 보지 못하는 상황이였고, 테이블의 규약을 맞춰 작성하자 잘 작동하였습니다.

감사합니다.

**Reply by laurenceisla** [reply]
Just copy pasting the Github automatic translation:

> Another kind person helped.
> 
> The problem was the protocol. Each table had its own rules, so I couldn't see them, and when I wrote them according to the rules of the table, it worked well.
> 
> I appreciate it.
